### PR TITLE
raise on invalid upload entry size

### LINKED ISSUE #1468: LiveViewTest render_submit raises unexpectedly with invalid upload size
### Environment

* Elixir version (elixir -v): `Elixir 1.11.1 (compiled with Erlang/OTP 23)`
* Phoenix version (mix deps): 1.5.9
* Phoenix LiveView version (mix deps): 0.15.7
* NodeJS version (node -v): v12.11.0
* NPM version (npm -v): 6.11.3
* Operating system: MacOS
* Browsers you attempted to reproduce this bug on (the more the merrier): n/a
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: n/a

### Actual behavior

If I provide an invalid `size` for an upload entry, a subsequent call to `render_submit()` will raise:

```elixir
test "uploads", %{conn: conn} do
  {:ok, lv, _html} = live(conn, "/upload")

  # contrived, but you get the idea
  %{size: size} = File.stat!("/path/to/foo.png")
  file = %{name: "foo.png", size: size * 2}

  assert lv
         |> file_input("#form", :file, [file])
         |> render_upload("foo.png") =~ "100%"
  
  assert lv
         |> form("#form")
         |> render_submit() =~ "upload complete"
end
```

```elixir
1) test handle bad size (DropsWeb.BasicUploadsLiveTest)
     test/drops_web/live/basic_uploads_live_test.exs:61
     ** (EXIT from #PID<0.414.0>) an exception was raised:
         ** (RuntimeError) cannot consume uploaded file that is still in progress
             (phoenix_live_view 0.15.7) lib/phoenix_live_view/upload_channel.ex:27: Phoenix.LiveView.UploadChannel.consume/3
             (elixir 1.11.1) lib/enum.ex:1399: Enum."-map/2-lists^map/1-0-"/2
             (drops 0.1.0) lib/drops_web/live/basic_uploads_live.ex:29: DropsWeb.BasicUploadsLive.handle_event/3
             (phoenix_live_view 0.15.7) lib/phoenix_live_view/channel.ex:342: anonymous fn/3 in Phoenix.LiveView.Channel.view_handle_event/3
             (telemetry 0.4.3) /Users/mcrumm/Code/mcrumm/live_upload_example/deps/telemetry/src/telemetry.erl:272: :telemetry.span/3
             (phoenix_live_view 0.15.7) lib/phoenix_live_view/channel.ex:204: Phoenix.LiveView.Channel.handle_info/2
             (stdlib 3.13.2) gen_server.erl:680: :gen_server.try_dispatch/4
             (stdlib 3.13.2) gen_server.erl:756: :gen_server.handle_msg/6
             (stdlib 3.13.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

### Expected behavior

Under test, we calculate the real byte_size of the given content so we can simulate the chunk behaviour. If the real size differs from the given size, I propose we raise an error letting the user know that the sizes differ, as this likely indicates a problem in the user's test, and the current error does not make that clear.

// @maennchen

### PR DESCRIPTION
**Pull request by mcrumm**

### CODE REVIEW

File: lib/phoenix_live_view/test/live_view_test.ex [hunk 1]
```diff
@@ -1413,7 +1413,7 @@ defmodule Phoenix.LiveViewTest do
         {:ok, %{ref: ref, config: config, entries: entries_resp}} ->
           case UploadClient.allowed_ack(upload, ref, config, entries_resp) do
             :ok -> render_chunk(upload, entry_name, percent)
-            {:error, reason} -> {:error, reason}
+            {:error, reason} -> raise "failed to render upload, reason: #{reason}"
```
Comment (chrismccord) [inline lib/phoenix_live_view/test/live_view_test.ex:1416]:
```suggestion
            {:error, reason} -> raise "failed to render upload, reason: #{inspect(reason)}"
```

File: lib/phoenix_live_view/test/upload_client.ex [hunk 2]
```diff
@@ -84,7 +79,38 @@ defmodule Phoenix.LiveViewTest.UploadClient do
...
+      } = client_entry
+
+      token = Map.fetch!(entries_resp, ref)
+      content_size = byte_size(content)
+
+      if content_size == client_size do
+        {:cont,
+         {:ok, Map.put(acc, name, build_and_join_entry(state, client_entry, content_size, token))}}
+      else
+        reason = """
```
Comment (chrismccord) [inline lib/phoenix_live_view/test/upload_client.ex:98]: can we simply raise here or is the issue this is running outside of the caller?

Comment (mcrumm) [reply lib/phoenix_live_view/test/upload_client.ex:98]: correct, but on second thought let's not do this at all 😂

**Comment by mcrumm** [comment]
On further consideration I will close this in favor of #1473 – there is further refactoring I'd like to do around preflight that will likely be complicated by this change, so at worst let's save it for later.

**Pull request closed without merging**